DigitalOffice/Services/CheckRightsService/Database/Entities/RightChangeRecordProjectLink.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace CheckRightsService.Database.Entities
+{
+    public class RightChangeRecordProjectLink
```

Comment:
Why do we need a separate class?

Reply:
Thanks for the dialogue in Slack, it became clear what you meant. I have a new question about Type. As I understand it, there may be the right to edit "X" (for example, the description for the project). And we will have to do RightToEditDescriptionToProject, maybe we should put the Type in a linking table?

Reply:
> Thanks for the dialogue in Slack, it became clear what you meant. I have a new question about Type. As I understand it, there may be the right to edit "X" (for example, the description for the project). And we will have to do RightToEditDescriptionToProject, maybe we should put the Type in a linking table?

Check out latest changes, I guess there're right types table u r talking about. If not, comment again :)

Reply:
I saw the following entities:
- DbRightType (Id, RightType Type, ICollection<RightTypeLink> Rights, ICollection<RightChangeRecordTypeLink> RightChangeRecords)
- Right(Id, UserId, ICollection<RightTypeLink> Types, ICollection<RightProjectLink> PermissionsIds)
- RightChangeRecord(Id, RightId, ChangedByUserId, DateTime Time, ICollection<RightChangeRecordTypeLink> Types, ICollection<RightRecordProjectLink> ChangedPermissionsIds)
- RightChangeRecordTypeLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, RightTypeId, DbRightType RightType)
- RightProjectLink(RightId, Right Right, ProjectId)
- RightRecordProjectLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, ProjectId)
- RightTypeLink(RightId, Right Right, RightTypeId, DbRightType RightType)
-  enum RightType 
And they got me confused. Maybe we should add a folder for the linking tables. Maybe we should change the naming convention. I propose to discuss this at the general call.

Reply:
And here the linking class convention is broken (Entity1Entity2 is a class linking the Entity1 and the Entity2)

Reply:
> I saw the following entities:
> 
> * DbRightType (Id, RightType Type, ICollection Rights, ICollection RightChangeRecords)
> * Right(Id, UserId, ICollection Types, ICollection PermissionsIds)
> * RightChangeRecord(Id, RightId, ChangedByUserId, DateTime Time, ICollection Types, ICollection ChangedPermissionsIds)
> * RightChangeRecordTypeLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, RightTypeId, DbRightType RightType)
> * RightProjectLink(RightId, Right Right, ProjectId)
> * RightRecordProjectLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, ProjectId)
> * RightTypeLink(RightId, Right Right, RightTypeId, DbRightType RightType)
> * enum RightType
> 
> And they got me confused. Maybe we should add a folder for the linking tables. Maybe we should change the naming convention. I propose to discuss this at the general call.

Ok np with discussion :)

Reply:
> I saw the following entities:
> 
> * DbRightType (Id, RightType Type, ICollection Rights, ICollection RightChangeRecords)
> * Right(Id, UserId, ICollection Types, ICollection PermissionsIds)
> * RightChangeRecord(Id, RightId, ChangedByUserId, DateTime Time, ICollection Types, ICollection ChangedPermissionsIds)
> * RightChangeRecordTypeLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, RightTypeId, DbRightType RightType)
> * RightProjectLink(RightId, Right Right, ProjectId)
> * RightRecordProjectLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, ProjectId)
> * RightTypeLink(RightId, Right Right, RightTypeId, DbRightType RightType)
> * enum RightType
> 
> And they got me confused. Maybe we should add a folder for the linking class. Maybe we should change the naming convention. I propose to discuss this at the general call.

Discussion is a good thing

Reply:
> And here the linking class convention is broken (Entity1Entity2 is a class linking the Entity1 and the Entity2)
> Maybe we should change the naming convention this way: Entity1_Entity2

Mb I messed up with following naming convension, not sure :)